aptos-move/move-examples/redemption/sources/redemption.move

```diff
@@ -0,0 +1,139 @@
+/// This module implements a simple redemption pool that allows users to redeem wrapped coins for native fungible assets.
+///
+/// A new pool can only be created by the deployer of the contract.
+/// Any operator can deposit native fungible assets into the pool.
+/// Operators can withdraw wrapped coins from the pool, up to the amount they have deposited.
+/// Users can redeem wrapped coins for native fungible assets at 1:1.
+module redemption::redemption {
+    use aptos_framework::aptos_account;
+    use aptos_framework::coin::{Self, Coin};
+    use aptos_framework::dispatchable_fungible_asset;
+    use aptos_framework::fungible_asset::{Self, Metadata};
+    use aptos_framework::object::{Self, Object, ExtendRef};
+    use aptos_framework::primary_fungible_store;
+    use aptos_std::table::{Self, Table};
+    use std::signer;
+    use std::string::String;
+    use aptos_std::type_info;
+    use aptos_framework::event;
+
+    /// Caller is not authorized to perform the operation.
+    const EUNAUTHORIZED: u64 = 1;
+
+    /// Operator cannot withdraw more than they have deposited.
+    const EBALANCE_EXCEEDED: u64 = 2;
+
+    struct RedemptionPool<phantom WrappedCoin> has key {
+        wrapped_coins: Coin<WrappedCoin>,
+        to: Object<Metadata>,
+        native_store: ExtendRef,
+        operator_balances: Table<address, u64>,
+    }
+
+    #[event]
+    struct CreatePool has drop, store {
+        coin: String,
+        to: Object<Metadata>,
+    }
+
+    #[event]
+    struct DepositNative has drop, store {
+        coin: String,
+        to: Object<Metadata>,
+        operator: address,
+        amount: u64,
+    }
+
+    #[event]
+    struct WithdrawWrapped has drop, store {
+        coin: String,
+        to: Object<Metadata>,
+        operator: address,
+        amount: u64,
+    }
+
+    #[event]
+    struct Redeem has drop, store {
+        coin: String,
+        to: Object<Metadata>,
+        user: address,
+        amount: u64,
+    }
+
+    public entry fun create_pool<WrappedCoin>(redemption_signer: &signer, to: Object<Metadata>) {
```

Comment:
nit: The usage of `to` seems off to me. `to` makes me think of the location of a store, rather than the metadata.
Might be better to rename to something like `metadata`?

Reply:
Or `redemption_fa`